packages/wallets/src/components/ModalProvider/ModalProvider.tsx

```diff
@@ -33,8 +37,9 @@ const ModalProvider = ({ children }: React.PropsWithChildren<unknown>) => {
     const modalRef = useRef<HTMLDivElement>(null);
     const [content, setContent] = useState<React.ReactNode | null>();
     const modalState = useRef<TModalState>();
-    const { isDesktop, isMobile } = useDevice();
+    const { isDesktop } = useDevice();
 
+    const [customRootRef, setCustomRootRef] = useState<RefObject<HTMLElement> | null>(null);
```

Comment:
Need to declare this as a state, once `customRootRef` is updated `modalRootRef` will be re-calculated